### PR TITLE
Tier-3 - CEPH-83575622 - cephfs_watchlist_bz

### PR DESCRIPTION
**Pull request by hkadam134**
# Description
    """
    CEPH-83575622 - create a file(fileA) , create a hard link (fileB),
        remove the fileA first and then fileB and run this in a loop for some time and check for deadlock
    Procedure :
    1. Create 2 Subvolumes - each mounted on kernel and fuse.
    2. Run unlink and rename to validate in any deadlock is seen.
    3. Check for Ceph Health Status to validate any warnings or errors if any deadlock is occured.

    Clean-up:
    1. Remove files from mountpoint, Unmount subvolumes.
    3. ceph fs subvolume rm <vol_name> <subvol_name> [--group_name <subvol_group_name>]
    4. ceph fs subvolumegroup rm <vol_name> <group_name>
    """

Polarion TC : https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83575622
Test Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-UI3ISC/

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83575622 - create a file(fileA) , create a hard link (fileB),
+        remove the fileA first and then fileB and run this in a loop for some time and check for deadlock
+    Procedure :
+    1. Create 2 Subvolumes - each mounted on kernel and fuse.
+    2. Run unlink and rename to validate in any deadlock is seen.
```
Comment (sumabai) [inline tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py:18]: to validate if any deadlock is seen.
change in to if to read as above

File: tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                    sudo=True,
+                    cmd=f"rm -rf {fuse_mounting_dir_1}{'primary_file_f' if i == 0 else f'hardlink{i}_file_f'}",
+                )
+        time.sleep(1)
+
+        log.info(
+            "Check for the Ceph Health to see if there are any deadlock bw unlink and rename."
+        )
+        ceph_health = fs_util_v1.get_ceph_health_status(clients[0])
+        log.info(ceph_health)
```
Comment (sumabai) [inline tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py:132]: we are not taking an action(raise exception) if ceph health is bad, is that ok?
Also, we are printing 'None' here, can we print actual health status into test log?

2023-09-12 01:31:27,132 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.tests.cephfs.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename.py:117 - Check for the Ceph Health to see if there are any deadlock bw unlink and rename.
2023-09-12 01:31:27,133 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.ceph.ceph.py:1558 - Running command ceph -s -f json on 10.0.211.12 timeout 600
2023-09-12 01:31:27,486 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.ceph.ceph.py:1592 - Command completed successfully
2023-09-12 01:31:27,487 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.tests.cephfs.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename.py:119 - None

**Review by sumabai — DISMISSED** [review]

**Comment by hkadam134** [comment]
Updated logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-SBPBDO/

File: tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py (hunk 2 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py]: Updated the script to raise exceptions.

File: tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py (hunk 1 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py]: Updated

**Review by AmarnatReddy — APPROVED** [review]
LGTM

**Review by AmarnatReddy — APPROVED** [review]
LGTM

**Pull request merged**